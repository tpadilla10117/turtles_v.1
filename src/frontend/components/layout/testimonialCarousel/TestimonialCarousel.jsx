import React, { useState, /* useEffect, useMemo */ } from 'react';
/* import { useInView } from 'react-intersection-observer'; */
import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';

function TestimonialCarousel( {carouselData} ) {

  /* CAROUSEL LOGIC: */

  //Using the carouselData index to matach against the number from currentSlide
  //If there's a match, the DOM elements (slide, buttons), get an 'active' class
  // Else, they get their normal names (**SEE TERNARY IN DOM ELEMENTS**)

  let [ currentSlide, setCurrentSlide ] = useState(0);
  const slideLength = carouselData.length;

/* Function to automatically update the carousel after a designated time with setTimeout, a DOM method: */
  /* useEffect( () => {
    const time = setTimeout( () => {
      setCurrentSlide(currentSlide === slideLength - 1 ? 0 : currentSlide + 1)  
    }, 10000);


    return () => {
      clearTimeout(time);
    }
  }, [currentSlide, slideLength]);
 */

/* Functions for Arrow Buttons if Requested by Client: */

  const nextSlide = () => {
    setCurrentSlide(currentSlide === slideLength - 1 ? 0 : currentSlide + 1);
    console.log('cliecked')
  };

  const previousSlide = () => {
    setCurrentSlide(currentSlide === 0 ? slideLength - 1 : currentSlide - 1);
  };



/* Helper Function that allows user to navigate to exact slide in carousel: */
  const exactSlide = (index) => {
    setCurrentSlide(parseInt(index));
  };

  const navigateDots = event => {
    //First need to target correct dot at the array index:
    const arrayValues = event.target.getAttribute('data-key');
    /* console.log('Here are my dots: ', arrayValues); */

    exactSlide(arrayValues);
  }

/* Intersection Observer Logic:
 */
 /*  const insersectionOptions = useMemo( ()=> {
    return {
      threshold: 1,
      root: null, 
      rootMargin: '0px 0px 00px 0px',
      triggerOnce: true
    }
   }, []);
    
    const { ref: clientImgRef, inView: isVisible, }= useInView(insersectionOptions); */


  return (
    <section className='testimonialCarousel-parent-container'>
      <ArrowForwardIosIcon className='testimonialCarousel-forwardarrow'
        onClick={nextSlide}
      />

      <ArrowBackIosNewIcon className='testimonialCarousel-backarrow'
        onClick={previousSlide}
      />
      
      <div className='testimonialCarousel-img-slider'>
        {carouselData.map( (img, index) => {
          return (
            <div className={index === currentSlide ? 'testimonialCarousel-slide active' : 'testimonialCarousel-slide'} key={img.id}>
              {/* <img alt={img.alt} src={img.src} /> */}

            {/* TODO: the client img */}
              <div className={index === currentSlide ? 'testimonialCarousel-clientimg-parent active fadeInUp' : 'testimonialCarousel-clientimg-parent'} /* ref={clientImgRef} */>
                <img alt={img.alt} src={img.clientimg} className='testimonialCarousel-clientimg' />
              </div>

              <div className={`testimonialCarousel-info ${img.id}`}>
              
                <h2 
                  className={`testimonialCarousel-title 
                  ${img.title}`}
                >
                  {img.title}
                </h2>

                <h3
                  className={`testimonialCarousel-subtitle 
                  ${img.subtitle}`}
                >
                  {img.subtitle}
                </h3>

                <p 
                  className={`testimonialCarousel-caption ${img.id}`}
                >
                  {img.caption}
                </p>

                {/* <hr></hr> */}

              </div>
            </div>
          )
        })}

        <div className='testimonialCarousel-navigation'>

          {carouselData.map( (btn, index) => {
            return (
              <div className={index === currentSlide ? 'testimonialCarousel-navigation-btn active' : 'testimonialCarousel-navigation-btn'} key={btn.buttonId} data-key={index} /* onClick={event => activeSlide(event)} */ onClick={navigateDots}></div>
            )
          })}

        </div>

      </div>

    </section>
  )
}

export default TestimonialCarousel;