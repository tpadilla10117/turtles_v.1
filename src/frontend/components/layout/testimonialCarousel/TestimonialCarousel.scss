.testimonialCarousel-parent-container {
    display: flex;
    position: relative;
  /*   align-items: center;
    justify-content: center; */
    width: 100%;
    height: 500px;
    max-width: 100%;
    max-height: 100%;
    background-color: $background-secondary;
    background: linear-gradient(180deg, rgba(0,0,0,0.1) 0%, rgba(0,0,0,0.1) 100% ), linear-gradient(190deg, rgba(0,0,0,0.1) 0%, transparent 100%);

    .testimonialCarousel-forwardarrow {
        position: absolute;
        top: 45%;
        right: 5%;
        transform: translate(-50%, -50%);
        color: #fff;
        z-index: 999;
    }

    .testimonialCarousel-backarrow {
        position: absolute;
        top: 45%;
        left: 10%;
        transform: translate(-50%, -50%);
        color: #fff;
        z-index: 999;
    }

    .testimonialCarousel-forwardarrow,
    .testimonialCarousel-backarrow {

        transition: all 250ms ease-in-out;
        filter: drop-shadow(0px 4px 4px rgb(0 0 0 / 50%) );
        /* filter for Material UI SVGs */
        

        &:hover {
            cursor: pointer;
            transform: scale3d(1.3, 1.3, 1.3);
        }
    }


    .testimonialCarousel-img-slider {
        position: relative;
        width: 100%;
        height: 100%;
        /* margin: 10px; */
        /* border: 10px solid yellow; */

        &:before {
            
                content: '';
                position: absolute;
                top: 0;
                left: 0;
                right: 0;
                bottom: 0;
                background: linear-gradient(180deg, rgba(0,0,0,0.1) 0%, rgba(0,0,0,0.1) 100% ), linear-gradient(190deg, rgba(0,0,0,0.1) 0%, transparent 100%);
                z-index: 2;
            
        }

        .testimonialCarousel-slide {
            z-index: 1;
            position: absolute;
            width: 100%;
            height: 100%;
            /* border: 10px solid red; */
            /* clip-path: circle(0% at 0 50%); */
            opacity: 0;

            &.active {
                /* clip-path: circle(100% at 50% 00%);
                transition: 1.5s;
                transition-property: clip-path; */
                opacity: 1;
                transition-duration: 1s;
                overflow: hidden;
            }
           

/* TODO: client img section */
            .testimonialCarousel-clientimg-parent {
                border: 2px solid #fff;
                border-radius: 50%;
                position: absolute;
                top: 20%;
                left: 50%;
                transform: translate(-50%, -50%);
                height: 150px;
                width: 150px;
                /* animation-name: rotateIn;
                transform-origin: center;
                animation-duration: 3s;

                &.rotateIn {
                    @keyframes rotateIn {
                        from {
                          transform: rotate3d(0, 0, 1, -200deg);
                          opacity: 0;
                        }
                      
                        to {
                          transform: translate3d(0, 0, 0);
                          opacity: 1;
                        }
                      }
                } */
                

                .testimonialCarousel-clientimg {
                    z-index: 2;
                    width: 100%;
                    height: 100%;
                    object-fit: cover;
                    object-position: center;
                    border-radius: 50%;
                }
            }

            img {
                z-index: 1;
                width: 100%;
                height: 100%;
                /* border-radius: 5px; */
                object-fit: cover;
                object-position: center;
                
            }

            .testimonialCarousel-info {
                position: absolute;
                top: 65%;
                left: 50%;
                transform: translate(-50%, -50%);
                /* padding: 15px 30px; */
                width: 100%;
                height: 300px;
                /* border: 1px solid red; */
                line-height: 25px;
                /* background: rgba(0, 0, 0, 0.5); */
                display: flex;
                flex-direction: column;
                align-items: center;
                
                
                h2 {
                    color: #fff;
                    letter-spacing: 1px;
                    text-transform: uppercase;
                    margin: 10px 0 10px 0;
                    text-align: center;
                }

                h3 {
                    color: #fff;
                    letter-spacing: 1px;
                    text-transform: uppercase;
                    margin: 10px 0 10px 0;
                    font-size: 14px;
                    text-align: center;
                }

                p {
                    color: #fff;
                    /* background: rgba(0, 0, 0, 0.5); */
                    font-size: 16px;
                    width: 50%;
                    padding: 10px;
                    border-radius: 4px;
                    height: 150px;
                   
                }

               /*  hr {
                    display: block;
                    width: 100%;
                    border: 1px inset;
                    overflow: hidden;
                    
                } */
                
            }
        }

        .testimonialCarousel-navigation {
            z-index: 2;
            position: absolute;
            /* border: 10px solid green; */
            display: flex;
            bottom: 30px;
            left: 50%;
            transform: translateX(-50%);

            .testimonialCarousel-navigation-btn {
                background: rgba(255, 255, 255, 0.5);
                width: 12px;
                height: 12px;
                margin: 10px;
                border-radius: 50%;
                border: .5px solid #fff;
                cursor: pointer;
                /* clip-path: circle(100% at 0 50%); */

                &.active {
                    background: $text-primary;
                    box-shadow: 0 0 2px rgba(0, 0, 0, 0.5);
                    /* clip-path: circle(200% at 100% 100%);
                    transition: 1.5s;
                    transition-property: clip-path; */
                    animation: ripple;
                    animation-duration: .7s;
                    border: none;
                    /* transition: 250ms opacity ease-in-out; */


                    @keyframes ripple {
                        from {
                            transform: scale(0);
                            opacity: 1;
                        }
                        to {
                            transform: scale(4);
                            opacity: 0;
                        }
                    }
    
                }

                &:hover {
                    background: $text-primary;
                    transition: .5s ease-in-out;
                }

                
            }
        }
    }
}

@media (max-width: 820px) {

    .testimonialCarousel-parent-container {

        .testimonialCarousel-img-slider {
            /* width: 600px; */
            /* height: 375px; */

            .testimonialCarousel-slide {
                .testimonialCarousel-info {
                    padding: 10px 25px;

                    h2 {
                        font-size: 35px;
                    }

                    p {
                        width: 75%;
                        /* font-size: 16px; */
                    }
                }
            }

            .testimonialCarousel-navigation {
                bottom: 25px;

                .testimonialCarousel-navigation-btn {
                    width: 10px;
                    height: 10px;
                    margin: 8px;
                }
            }
        }
    }
};

@media (max-width: 620px) {

    .testimonialCarousel-parent-container {

        .testimonialCarousel-img-slider {
            /* width: 400px; */
            /* height: 250px; */

            .testimonialCarousel-slide {
                .testimonialCarousel-info {
                    padding: 10px 20px;

                    h2 {
                        font-size: 30px;
                    }

                    p {
                        width: 100%;
                        font-size: 16px;
                    }
                }
            }

            .testimonialCarousel-navigation {
                bottom: 15px;

                .testimonialCarousel-navigation-btn {
                    width: 8px;
                    height: 8px;
                    margin: 6px;
                }
            }
        }
    }
};

@media (max-width: 420px) {

    .testimonialCarousel-parent-container {

        .testimonialCarousel-img-slider {
            /* width: 400px; */
            height: 500px;

            .testimonialCarousel-slide {
                .testimonialCarousel-info {
                    padding: 5px 10px;

                    h2 {
                        font-size: 25px;
                    }

                    p {
                        width: 75%;
                        font-size: 11px;
                    }
                }
            }

            .testimonialCarousel-navigation {
                bottom: 10px;

            }
        }
    }
};