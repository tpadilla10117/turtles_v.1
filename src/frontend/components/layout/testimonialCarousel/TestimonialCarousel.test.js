import { shallow, mount } from 'enzyme';
import React from 'react';
import TestimonialCarousel from './TestimonialCarousel';

describe('The TestimonialCarousel.jsx component:', () => {

    const mockPhotoProps = [
        {
            src: 'picture.jpeg',
            title: 'Kianna B.',
            clientimg: 'picture.jpeg',
            subtitle: 'Student',
            caption: 'Turtles really is an amazing talent & teacher.  I have learned so much from here and would not be where I am today without her. Turtles really is an amazing talent & teacher.  I have learned so much from here and would not be where I am today without her',
            alt: 'Turtles Swinging in the studio',
            thumbnail: 'Turtles Swinging in the studio',
            buttonId: 1,
            id: 1
        }
    ]

    it('renders the comoponent', () => {
        expect.assertions(2);
        expect(shallow(<TestimonialCarousel carouselData={mockPhotoProps}/>).length).toEqual(1);
        expect(shallow(<TestimonialCarousel carouselData={mockPhotoProps} />)).toMatchSnapshot();
    })
})