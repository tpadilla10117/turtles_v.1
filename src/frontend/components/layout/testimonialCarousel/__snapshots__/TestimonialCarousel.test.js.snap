// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TestimonialCarousel.jsx component: renders the comoponent 1`] = `ShallowWrapper {}`;
